### DISCUSSION TITLE
Proposal: <podcast:episodeType>

### DISCUSSION THREAD

**Discussion by theDanielJLewis** [Podcastindex-org/podcast-namespace#470] [Tag Proposal]
I'm preparing an episode of _The Audacity to Podcast_ to teach about the "full," "bonus," and "trailer" episode types and when to use them. And I thought it would be good for use to include a drop-in replacement tag for `<itunes:episodeType>` since it pairs very well with episode numbers and seasons.

Thus, the `<podcast:episodeType>` tag, but with some improvements.

Apple explains their tag as follows:

> The episode type.
>
> If an episode is a trailer or bonus content, use this tag.
> 
> Where the episodeType value can be one of the following:
> 
> - Full (default). Specify full when you are submitting the complete content of your [show](https://help.apple.com/itc/podcasts_connect/#/itca5dea085b).
> 
> - Trailer. Specify trailer when you are submitting a short, promotional piece of content that represents a preview of your current show.
> 
> - Bonus. Specify bonus when you are submitting extra content for your show (for example, behind the scenes information or interviews with the cast) or cross-promotional content for another show.
> 
> The rules for using trailer and bonus tags depend on whether the <itunes:season> and <itunes:episode> tags have values:
> 
> Trailer:
> 
> - No season or episode number: a show trailer
> 
> - A season number and no episode number: a season trailer. (Note: an episode trailer should have a different <guid> than the actual episode)
> 
> - Episode number and optionally a season number: an episode trailer/teaser, later replaced with the actual episode
> 
> Bonus:
> 
> - No season or episode number: a show bonus
> 
> - A season number: a season bonus
> 
> - Episode number and optionally a season number: a bonus episode related to a specific episode

The resulting tag would look like `<itunes:episodeType>trailer</itunes:episodeType>`

The most important takeaway is that these bonus and trailer types are connected to something: the whole show, a season, or an episode.

I think our tag should function the same with this information, and add some additional uses.

- **Extra** or **special**: This would be the type for any kind of extra episode that's a break from the norm, but is not a bonus for existing content. For example, if my show is a weekly solo show, but I publish a bunch of 5-minute interviews from an event, those interviews would _not_ be the bonus type, because they're not attached to any existing content. They're standalone, but _extra_, and thus the "extra" type would be more appropriate than "bonus." This would give podcasters the episode type I often see them incorrectly thinking "bonus" is for.
- **Rerun**: This would indicate that the episode is republishing old content—either in whole or multiple segments.
- **Sequel** or **miniseries**: That might not be the right word, but here's what I envision. This type indicates that an episode is a _part_ of a miniseries regardless of the whole show type. Consider some TV shows, for example. The show is an overall serial format, but then there's a part 1 and part 2 miniseries. In other words, a small series within a series. But with out bonus and trailer types are attached to an episode/season number, I think this tag would stand alone, allowing "part 1" to be episode 10 and "part 2" to be episode 11. What happens when there are multiple miniseries and how to group them? That could _probably_ be handled with our season tag. But the following is another possible solution.

**Additional label attribute?**

While the Apple tag allows only a single value, I think our tag should have an optional "label" attribute, which can display to the audience as well as help group miniseries together.

For example, `<podcast:episodeType label="Foundations">miniseries</podcast:episodeType>`

This could be used when a particular season is already necessary, but the miniseries spreads across seasons or is independent of them.

But I'll admit this is probably an extremely rare scenario and nearly every situation I consider could be handled by `<podcast:season>` instead.

For example, many pastors will preach sermon series, but sometimes take a break from that series for holiday themes or summer. So a "Foundations" sermon series could be indicated in the season tag, while the episodes are assigned to it as normal episodes. But the "miniseries" episode type could be for a series too small to warrant its own season.

**Support multiple types?**

I can foresee the need for an episode to be multiple types. Like rerunning a past bonus episode. Or maybe something is a trailer for a miniseries. So it might be necessary to allow multiple instances of the episodeType tag.

**Comment by theDanielJLewis** [comment]
Added a "rerun" type, [thanks to inspiration from thejavakat](https://podcastindex.social/@thejavakat/109122311337078012).

**Comment by theDanielJLewis** [comment]
It was pointed out to me that we're [allowing decimals in episode numbers](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#episode) to allow for the things I think episode types can solve better.

**Comment by theDanielJLewis** [comment]
I thought of an additional episode type: `promo` or `promotion`. This can be used for an episode that serves only to promote something, like a special event, another podcast, a sale, and such.

This kind of content shouldn't be a trailer or bonus because it might not be tied to the rest of the show (it could be promoting a different show or something that's not even a podcast). And it's not a "special episode" either.

Promos would be the kind of content a podcaster might want to remove after the event has happened. But they might also opt to leave the promo in the feed indefinitely, such as when crosspromoting another podcast (of their own or someone else's).

This would pair perfectly with my new proposal for a `<podcast:expire>` tag (#448) to make episodes (of any type) automatically dim or hide from podcast apps and directories after a time is past.

**Comment by genebean** [comment]
I like the idea of more parity with the iTunes tags, but also caution against getting too granular. In reading this just now it seemed a bit like splitting hairs. You can almost always find smaller groups to categorize things as, but at a certain point it becomes too granular to be useful. I’m not sure this has made it to that stage yet, but it certainly seems on its way. 

Do also keep in mind the UI a host would need to present so the type could be specified. One thing that instantly came to mind when thinking about this is that a host won’t want to have two places this is entered so a mapping from all new types to the appropriate iTunes type is likely needed. 

For example: say I am in Buzzsprout’s web interface uploading an episode that’s not the default “full”. I’d see a list of these types and have to pick one after reading some **unambiguous** help text. 

- How would we describe each type in a way that creators easily understand?
- Under the hood, which value would be simultaneously inserted for `<itunes:episodeType>`?
- At what point are so many choices available in the UI that the average creator doesn’t really know which one to pick?
